api/v1alpha2/bundledeployment_types.go

```diff
@@ -69,8 +69,42 @@ type BundleDeploymentSpec struct {
 	//
 	// config is provisioner specific configurations
 	Config runtime.RawExtension `json:"config,omitempty"`
+
+	//+kubebuilder:Optional
+	// Preflight defines the strategy for the preflight checks (i.e. as of now, the CRD upgrade safety checks) to be enabled or disabled when attempting to install the BundleDeployment.
+	Preflight *PreflightConfig `json:"preflight,omitempty"`
+}
+
+// PreflightConfig holds the config for the preflight checks. Currently, this is implemented for the
+// CRDUpgradeSafety preflight check and can be extended in the future to add other preflight checks.
+type PreflightConfig struct {
+	//+kubebuilder:Optional
+	CRDUpgradeSafety *CRDUpgradeSafetyPreflightConfig `json:"crdUpgradeSafety,omitempty"`
+}
+
+// CRDUpgradeSafetyPreflightConfig is a custom struct that holds the necessary configuration values to
+// enable or disable the CRD Upgrade Safety validations. Currently, this holds Mode
+// that sets the CRD Upgrade Safety validations to either Enabled/Disabled.
+// By default, this is set to "Enabled".
+type CRDUpgradeSafetyPreflightConfig struct {
+	//+kubebuilder:validation:Enum:=Enabled;Disabled
+	//+kubebuilder:default:=Enabled
+	//+kubebuilder:Optional
+	Mode *CRDUpgradeSafetyMode `json:"mode,omitempty"`
```

Comment:
Do we envision add more modes here (e.g. to enable/disable certain facets of the check)? Or are we thinking that would end up going into a separate field or fields?

Reply:
Awhile back with `.spec.paused`, I flip-flopped on "to bool, or not to bool" and I ended up feeling like a bool actually worked better. I wonder if that's the case here as well.

Reply:
I was thinking to keep this an enum with the idea that this could be used for modes like Enabled, Disabled, Strict, Loose.

We had originally discussed a potential in the future to have designated Strict/Loose options but if we don't think that is something that would be worth pursuing in the long run then I would be okay swapping this back to a bool

Reply:
> Do we envision add more modes here (e.g. to enable/disable certain facets of the check)? Or are we thinking that would end up going into a separate field or fields?

I think this would end up going into a separate field or a set of fields. 
Also, I renamed `Mode` to `Enabled` and it's a bool now - that either enables/disables the entire preflight check.